extensions/eclipse-che-theia-user-preferences/src/browser/che-frontend-application.ts

```diff
@@ -0,0 +1,70 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+import {
+  ApplicationShell,
+  FrontendApplication,
+  FrontendApplicationContribution,
+  KeybindingRegistry,
+  ShellLayoutRestorer,
+} from '@theia/core/lib/browser';
+import { CommandRegistry, ContributionProvider, MenuModelRegistry } from '@theia/core';
+import { inject, injectable, named } from 'inversify';
+
+import { CheShellLayoutRestorer } from './che-shell-layout-restorer';
+import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
+import { StorageServicePreferences } from './che-storage-preferences';
+
+@injectable()
+export class CheFrontendApplication extends FrontendApplication {
```

Comment:
Is there any good reason why we need to rebind the upstream component and depend on it?
Can we register the required listeners by providing [`FrontendApplicationContribution`](https://github.com/eclipse-theia/theia/blob/42042974c59f148974de5e8fd90cba7222cee82f/packages/core/src/browser/frontend-application.ts#L33)?

Reply:
Theoretically it can be realized through the `onDidInitializeLayout` implementation. Need to try.

Reply:
Reimplemented

Reply:
Thanks!